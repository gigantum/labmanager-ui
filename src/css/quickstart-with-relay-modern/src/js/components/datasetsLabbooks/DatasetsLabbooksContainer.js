import React, { Component } from 'react'

export default class DatasetsLabbooksContainer extends Component {
  constructor(props){
    super(props);
    this.state = {
      selectedComponent: 'datasets'
    }
  }
  componentWillMount() {
;
  }

  setSelectedComponent(that, component){
    this.setState({selectedComponent: component})
  }
  render() {

    return (

      <div className='datasets-labbooks__container flex justify-center'>
        <ul className='datasets-labbooks__nav flex flex--row justify--space-between'>
          <li onClick={() => this.setSelectedComponent(this, 'datasets')} className={this.state.selectedComponent === 'datasets' ? 'datasets-labbooks__nav-item selected': 'datasets-labbooks__nav-item'}>Datasets</li>
          <li onClick={() => this.setSelectedComponent(this, 'labbooks')} className={this.state.selectedComponent === 'labbooks' ? 'datasets-labbooks__nav-item selected': 'datasets-labbooks__nav-item'}>Labbooks</li>
        </ul>

        {
          //(this.state.selectedComponent === 'datasets') ? return(<div> datasets </div>) : return(<div> labboks </div>)
        }
      </div>
    )
  }
}
