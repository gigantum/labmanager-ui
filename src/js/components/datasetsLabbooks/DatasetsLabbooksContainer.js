import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import {graphql, QueryRenderer} from 'react-relay'
//components
import DatasetSets from './datasets/DatasetSets';
import LabbookSets from './labbooks/LabbookSets';
import environment from './../../createRelayEnvironment'
import WizardModal from './../wizard/WizardModal'


const LabbookQuery = graphql`query DatasetsLabbooksContainerQuery($first: Int!, $cursor: String){
  localLabbooks(first:$first, after: $cursor) {
    ...LabbookSets_localLabbooks
  }
}`

export default class DatasetsLabbooksContainer extends Component {
  constructor(props){

    super(props);
    this.state = {
      selectedComponent: props.match.params.id
    }
  }

  _setSelectedComponent(that, component){
    this.setState({selectedComponent: component})
    this.props.history.push(`../${component}`)
  }

  _displaySelectedComponent(){
    if(this.state.selectedComponent === 'datasets'){

      return (<DatasetSets />)
    }else{

      return (<QueryRenderer
        environment={environment}
        query={LabbookQuery}
        variables={{
          first: 10,
          cursor: null
        }}

        render={({error, props}) => {

          if (error) {

            return <div>{error.message}</div>
          } else if (props) {

            return (

              <LabbookSets history={this.props.history} {...props}/>
            )
          }else{

            return (
              <div>
                <WizardModal
                  handler={this.handler}
                  history={this.props.history}
                  {...this.props}
                />
              </div>
            )
          }
        }}
      />
    )

    }
  }
  render() {

    return (
      <div className='DatasetsLabbooks flex flex-column'>
        <div className='DatasetsLabbooks__nav-container flex justify-center flex-0-0-auto'>
          <ul className='DatasetsLabbooks__nav flex flex--row justify--space-between'>
            <li>
              <Link
                onClick={(t,event) => this._setSelectedComponent(this, 'datasets')}
                className={this.state.selectedComponent === 'datasets' ? 'DatasetsLabbooks__nav-item selected': 'DatasetsLabbooks__nav-item'}
                to='../datasets'
              >
                Datasets
              </Link>
            </li>
            <li>
              <Link
                onClick={(t, event) => this._setSelectedComponent(this, 'labbooks')}
                className={this.state.selectedComponent === 'labbooks' ? 'DatasetsLabbooks__nav-item selected': 'DatasetsLabbooks__nav-item'}
                to='../labbooks'
              >
                Labbooks
              </Link>
            </li>

          </ul>
        </div>
        <div className='DatasetsLabbooks__view flex-1-0-auto'>
          {
            this._displaySelectedComponent()
          }
        </div>
      </div>
    )
  }
}
