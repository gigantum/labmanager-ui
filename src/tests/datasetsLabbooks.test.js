import React from 'react'
import renderer from 'react-test-renderer';
import {StaticRouter, Link} from 'react-router';
import DatasetsLabbooksContainer from './../js/components/datasetsLabbooks/DatasetsLabbooksContainer';

test('DatasetsLabbooksContainer Renders Correctly', () => {
  const context = {}
  const component = renderer.create(
    <StaticRouter location="home/datasets" context={context}>
      <DatasetsLabbooksContainer match={{params: {id: 'datasets'}}} />
    </StaticRouter>
  );
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
  console.log(tree.children[0].children[0].children[0].props)
  // manually trigger the callback
  tree.children[0].children[0].children[0].props.onClick()
  //re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.children[0].children[1].children[0].props.onClick()
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});
