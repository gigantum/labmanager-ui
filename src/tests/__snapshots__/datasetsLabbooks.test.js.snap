// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatasetsLabbooksContainer Renders Correctly 1`] = `
<div
  className="datasets-labbooks__container flex justify-center"
>
  <ul
    className="datasets-labbooks__nav flex flex--row justify--space-between"
  >
    <li
      className="datasets-labbooks__nav-item selected"
      onClick={[Function]}
    >
      Datasets
    </li>
    <li
      className="datasets-labbooks__nav-item"
      onClick={[Function]}
    >
      Labbooks
    </li>
  </ul>
</div>
`;

exports[`DatasetsLabbooksContainer Renders Correctly 2`] = `
<div
  className="datasets-labbooks__container flex justify-center"
>
  <ul
    className="datasets-labbooks__nav flex flex--row justify--space-between"
  >
    <li
      className="datasets-labbooks__nav-item selected"
      onClick={[Function]}
    >
      Datasets
    </li>
    <li
      className="datasets-labbooks__nav-item"
      onClick={[Function]}
    >
      Labbooks
    </li>
  </ul>
</div>
`;

exports[`DatasetsLabbooksContainer Renders Correctly 3`] = `
<div
  className="datasets-labbooks__container flex justify-center"
>
  <ul
    className="datasets-labbooks__nav flex flex--row justify--space-between"
  >
    <li
      className="datasets-labbooks__nav-item"
      onClick={[Function]}
    >
      Datasets
    </li>
    <li
      className="datasets-labbooks__nav-item selected"
      onClick={[Function]}
    >
      Labbooks
    </li>
  </ul>
</div>
`;

exports[`DatasetsLabbooksContainer Reners Correctly 1`] = `
<div
  className="datasets-labbooks__container flex justify-center"
>
  <ul
    className="datasets-labbooks__nav flex flex--row justify--space-between"
  >
    <li
      className="datasets-labbooks__nav-item selected"
      onClick={[Function]}
    >
      Datasets
    </li>
    <li
      className="datasets-labbooks__nav-item"
      onClick={[Function]}
    >
      Labbooks
    </li>
  </ul>
</div>
`;

exports[`DatasetsLabbooksContainer Reners Correctly 2`] = `
<div
  className="datasets-labbooks__container flex justify-center"
>
  <ul
    className="datasets-labbooks__nav flex flex--row justify--space-between"
  >
    <li
      className="datasets-labbooks__nav-item selected"
      onClick={[Function]}
    >
      Datasets
    </li>
    <li
      className="datasets-labbooks__nav-item"
      onClick={[Function]}
    >
      Labbooks
    </li>
  </ul>
</div>
`;

exports[`DatasetsLabbooksContainer Reners Correctly 3`] = `
<div
  className="datasets-labbooks__container flex justify-center"
>
  <ul
    className="datasets-labbooks__nav flex flex--row justify--space-between"
  >
    <li
      className="datasets-labbooks__nav-item"
      onClick={[Function]}
    >
      Datasets
    </li>
    <li
      className="datasets-labbooks__nav-item selected"
      onClick={[Function]}
    >
      Labbooks
    </li>
  </ul>
</div>
`;
